### PR TITLE
Gh 54/johnnyhchen

### LINKED ISSUE #54: [AWS Amplify: Datastore] Create a new GraphQL schema to match existing database schema structure
Storypoints: 2

This might be as simple as copy and pasting the existing schema definitions and adding a couple tags.

This should exist in the src/db folder.

Should only give users read access, as write access operations should still be done through APIs for now.

Context: https://github.com/gp-seattle/rms-app/wiki/AWS-Amplify-Project:-Datastore

### PR DESCRIPTION
**Pull request by johnnyhchen**
Resolves #54

### CODE REVIEW

File: amplify/backend/api/rms/schema.graphql [hunk 1]
```diff
@@ -1,5 +1,76 @@
...
+    allItems(count: Int, nextToken: String): PaginatedItems!
+    allBatch(count: Int, nextToken: String): PaginatedBatches!
+    allTags(count: Int, nextToken: String): PaginatedTags!
+    allHistory(count: Int, nextToken: String): PaginatedHistories!
 }
+
+type Main @model @auth(rules: [{allow: public, operations: [read]}]) {
+    name: ID!
+    displayName: String!
+    description: String!
```
Comment (yauj) [inline amplify/backend/api/rms/schema.graphql:12]: This is optional

File: amplify/backend/api/rms/schema.graphql [hunk 2]
```diff
@@ -1,5 +1,76 @@
...
+
+type Items @model @auth(rules: [{allow: public, operations: [read]}]) {
+    id: ID!
+    name: String!
+    owner: String!
+    borrower: String!
+    batch: [String!]
+    history: [String!]
+    schedule: [String!]
+    notes: String!
```
Comment (yauj) [inline amplify/backend/api/rms/schema.graphql:30]: This is optional

File: amplify/backend/api/rms/schema.graphql [hunk 3]
```diff
@@ -1,5 +1,76 @@
...
+    tags: [Tags!]!
+    nextToken: String
+}
+
+type History @model @auth(rules: [{allow: public, operations: [read]}]) {
+    key: ID!
+    name: String
+    id: String
+    borrower: String
+    action: Action
```
Comment (yauj) [inline amplify/backend/api/rms/schema.graphql:63]: These are required

File: amplify/backend/api/rms/schema.graphql [hunk 4]
```diff
@@ -1,5 +1,76 @@
...
+    notes: String
+    timestamp: Int
```
Comment (yauj) [inline amplify/backend/api/rms/schema.graphql:65]: This is required

**Review by yauj — CHANGES_REQUESTED** [review]

File: amplify/backend/api/rms/schema.graphql (hunk 1 above)
Comment (johnnyhchen) [reply amplify/backend/api/rms/schema.graphql:12]: Optional as in don't use "!" (nullable) or as in remove these fields?

**Review by yauj — APPROVED** [review]

**Comment by johnnyhchen** [comment]
Desirable
- GraphQL Schema defined correctly and set up as an AppSync API
- DynamoDB Tables properly added as datasources in AppSync 
- Get whole table queries will work and are connected to the proper resolver & DynamoDB DataSource
Neutral
- Autogenerated DynamoDB tables still exist in DynamoDB
- - Possible fix: Can delete these using console but might come back every time a new Amplify CD deployment script is run
- The permissions given in the PolicyDocument section of CustomResources.json could possibly be refined. Can look into which Actions are necessary to pull a table from DynamoDB and add it to AppSync as a DataSource
Not Desirable
- All other queries do not have the proper resolver set up and will query from the auto-generated/wrong table
 - - Possible fix: Workaround exists by deleting the datasource using console and pushing the resolvers with the correct datasource. Would need to do another push. See https://github.com/aws-amplify/amplify-cli/issues/682#issuecomment-490463285 and https://forums.aws.amazon.com/thread.jspa?messageID=850754
- Not sure if string sets will be returned for certain desired fields or string arrays. We want string sets. Need to verify.

**Pull request merged**